lib/internal/fs/streams.js

```diff
@@ -251,6 +272,32 @@ function WriteStream(path, options) {
   this.flags = options.flags === undefined ? 'w' : options.flags;
   this.mode = options.mode === undefined ? 0o666 : options.mode;
 
+  if (options.open && typeof options.open !== 'function') {
+    throw new ERR_INVALID_ARG_TYPE('open', 'function', options.open);
```

Comment:
Since these are passed in a options, the first argument for each of the relevant `ERR_INVALID_ARG_TYPE` constructors should follow the pattern `'options.open'`
```suggestion
    throw new ERR_INVALID_ARG_TYPE('options.open', 'function', options.open);
```